### ISSUE TITLE
Perform spurious wakeups from `park`?

### ISSUE THREAD

**Issue by RalfJung** [rust-lang/miri#2307]
The [specification of `park`](https://doc.rust-lang.org/nightly/std/thread/fn.park.html) allows spurious wakeups. I am not entirely sure what our implementation of the underlying OS primitive currently entails, but it might be that it will never spuriously wake up. In which case it'd be nice if we could somehow artificially make it wake up -- like with compare_exchange_weak, that is probably a not-well-tested code path in programs using `park`.

**Comment by bjorn3** [comment]
At least `park_timeout` can have spurious wakeups in practice on linux as there is no way to distinguish a spurious wakeup and a timeout: https://github.com/rust-lang/rust/blob/baaa3b682986879c7784b5733ecea942e9ae7de3/library/std/src/sys_common/thread_parker/futex.rs#L78-L80 On macOS `park_timeout` will spuriously wakeup after 1000 years to work around an OS bug: https://github.com/rust-lang/rust/blob/baaa3b682986879c7784b5733ecea942e9ae7de3/library/std/src/sys/unix/thread_parker.rs#L64-L66 On windows `park_timeout` can also spuriously wakeup for the same reason as linux: https://github.com/rust-lang/rust/blob/baaa3b682986879c7784b5733ecea942e9ae7de3/library/std/src/sys/windows/thread_parker.rs#L158-L160

**Comment by RalfJung** [comment]
Right, but can that actually happen with the Miri implementation of `futex_wait`?
